src/finiteVolume/cellCentred/operators/gaussGreenDiv.cpp

```diff
@@ -156,32 +156,36 @@ void computeDivImp(
     la::LinearSystem<ValueType, localIdx>& ls,
     const SurfaceField<scalar>& faceFlux,
     const VolumeField<ValueType>& phi,
+    const SurfaceInterpolation<ValueType>& surfInterp,
     const dsl::Coeff operatorScaling,
     const SparsityPattern& sparsityPattern
 )
 {
     const UnstructuredMesh& mesh = phi.mesh();
     const auto nInternalFaces = mesh.nInternalFaces();
     const auto exec = phi.exec();
+    const auto weights = surfInterp.weight(faceFlux, phi);
 
-    const auto [sFaceFlux, owner, neighbour, surfFaceCells, diagOffs, ownOffs, neiOffs] = views(
-        faceFlux.internalVector(),
-        mesh.faceOwner(),
-        mesh.faceNeighbour(),
-        mesh.boundaryMesh().faceCells(),
-        sparsityPattern.diagOffset(),
-        sparsityPattern.ownerOffset(),
-        sparsityPattern.neighbourOffset()
-    );
+    const auto [sFaceFlux, sWeights, owner, neighbour, surfFaceCells, diagOffs, ownOffs, neiOffs] =
```

Comment:
maybe we should start adding a V for view to the end.  

```suggestion
    const auto [sFaceFlux, weightsV, owner, neighbour, surfFaceCells, diagOffs, ownOffs, neiOffs] =
```